src/lib.rs

```diff
@@ -551,6 +555,52 @@ mod tests {
         Ok((collection, items, stats))
     }
 
+    // helper to aggregate a get response and return all relevant data
```

Comment:
let's nitpick: Always write doc comments in proper style: 1 sentence summary with capitals and punctuation in imperative style, optionally followed by a blank line and more explanation.  That way it renders nicely with `cargo doc --document-private-items`.